app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt

```diff
@@ -93,13 +93,14 @@ class PatientsScreenController @Inject constructor(
         // So duplicate triggers are dropped by restricting flatMap's
```

Comment:
If you look at the `PatientsScreenController#refreshUserStatus` method, we can see that a complete data sync happens whenever the user gets refreshed if they are logged in.

This was meant to handle the case where sync should happen as soon as a user is approved. Now with this change, a complete sync will happen every time the screen is opened.

In order to maintain the current functionality, we should do this sync only whenever the user's logged in status and user status changes to (`Allowed`, Logged In`) from any other state.

Reply:
I will raise a new PR for this fix.